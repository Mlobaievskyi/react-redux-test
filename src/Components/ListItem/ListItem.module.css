.container {
  display: flex;
  justify-content: space-between;
  align-items: center;

  padding: 20px;
  border-radius: 5px;
  box-shadow: var(--shadow) var(--shadow-color);
  cursor: pointer;

  margin-bottom: 20px;
  border-left: 2px solid var(--white-color);
}

.selectedItem {
  border-left: 2px solid var(--main-color);
}

.leftSide {
  display: flex;
  align-items: center;
}

.radioBtn {
  padding: 20px;
}

.info {
  display: flex;
  flex-direction: column;
  margin-left: 20px;
}

.name {
  font-size: 18px;
  font-weight: bold;

  margin-bottom: 10px;
}