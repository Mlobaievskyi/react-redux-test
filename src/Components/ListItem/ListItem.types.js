// @flow

import React from "react";

export type ListItemProps = {|
  name: string,
  timestamp: Date,
  selected: boolean,
  actions?: React.Children<>,
  selectItem: () => void,
|};
