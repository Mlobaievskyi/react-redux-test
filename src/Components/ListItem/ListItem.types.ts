import type React from "react";

export type ListItemProps = {
  name: string;
  timestamp: Date;
  selected: boolean;
  actions?: React.ReactNode;
  selectItem: () => void;
};
