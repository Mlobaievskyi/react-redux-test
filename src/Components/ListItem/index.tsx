import React from "react";
import dayjs from "dayjs";
import { Radio } from "antd";
import cn from "classnames";

import type { ListItemProps } from "./ListItem.types";
import s from "./ListItem.module.css";

export default ({
  name,
  timestamp,
  selected,
  selectItem,
  actions,
}: ListItemProps) => (
  <div
    className={cn(s.container, selected && s.selectedItem)}
    onClick={selectItem}
  >
    <div className={s.leftSide}>
      <Radio checked={selected} />
      <div className={s.info}>
        <span className={s.name}>{name}</span>
        <span>{dayjs(timestamp).format("MMM-DD mm:ss")}</span>
      </div>
    </div>

    <div className={s.actions}>{actions}</div>
  </div>
);
