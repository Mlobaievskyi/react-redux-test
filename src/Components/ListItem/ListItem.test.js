// @flow

import React from "react";
import ListItem from "./";
import cn from "classnames";
import dayjs from "dayjs";
import { shallow } from "enzyme";

const mockedListItem = {
  name: "Test name",
  timestamp: new Date(2020, 1, 1),
  id: 1,
};

jest.mock("classnames");
jest.mock("dayjs");

describe("ListItem component tests", () => {
  beforeEach(() => {
    cn.mockReturnValue("testClassName");
    dayjs.mockReturnValue({ format: () => "today" });
  });

  afterEach(() => {
    cn.mockReset();
    dayjs.mockReset();
  });

  test("renders itself", () => {
    const component = shallow(
      <ListItem
        name={mockedListItem.name}
        timestamp={mockedListItem.timestamp}
        selected={false}
        selectItem={() => {}}
      />
    );

    expect(component).toHaveLength(1);
    component.unmount();
  });

  it("renders proper data", () => {
    const component = shallow(
      <ListItem
        name={mockedListItem.name}
        timestamp={mockedListItem.timestamp}
        selected={false}
        selectItem={() => {}}
      />
    );

    const spans = component.find("span");
    const name = spans.at(0).prop("children");
    const date = spans.at(1).prop("children");

    expect(name).toEqual(mockedListItem.name);
    expect(date).toEqual("today");
    expect(component.hasClass("testClassName")).toBe(true);

    component.unmount();
  });
});
