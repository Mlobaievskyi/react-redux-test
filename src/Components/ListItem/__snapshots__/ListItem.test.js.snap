// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component tests Link changes the class when hovered 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <div
    className="leftSide"
  >
    <label
      className="ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
    </label>
    <div
      className="info"
    >
      <span
        className="name"
      >
        Test name
      </span>
      <span>
        Feb-01 00:00
      </span>
    </div>
  </div>
  <div
    className="actions"
  />
</div>
`;

exports[`ListItem component tests Link changes the class when hovered 2`] = `
<div
  className="container"
  onClick={[Function]}
>
  <div
    className="leftSide"
  >
    <label
      className="ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
    </label>
    <div
      className="info"
    >
      <span
        className="name"
      >
        Test name
      </span>
      <span>
        Feb-01 00:00
      </span>
    </div>
  </div>
  <div
    className="actions"
  />
</div>
`;
